Capri.Services/ServiceResult.cs

```diff
@@ -5,33 +5,52 @@
 
 namespace Capri.Services
 {
-    public class ServiceResult<T> : Dictionary<string, string>
+    public class ServiceResult<T> : Dictionary<string, string>, IServiceResult<T>
     {
-        public T TBody { get; private set; }
-        public bool IsSucceded { get; private set; }
+        private T _tBody;
+        private bool? _successful;
 
-        private ServiceResult() { }
+        public ServiceResult()
+        {
+        }
 
         public static ServiceResult<T> Success(T body)
         {
-            return new ServiceResult<T>
-            {
-                TBody = body,
-                IsSucceded = true
-            };
+            ServiceResult<T> serviceResult = new ServiceResult<T>(true);
+            serviceResult._tBody = body;
+            return serviceResult;
         }
 
         public static ServiceResult<T> Error(params string[] errors)
         {
-            ServiceResult<T> serviceResult = new ServiceResult<T>();
-            serviceResult.IsSucceded = false;
-            foreach(var errorMessage in errors)
+            ServiceResult<T> serviceResult = new ServiceResult<T>(false);
+            foreach (var errorMessage in errors)
             {
                 serviceResult.AddError("error", errorMessage);
             }
             return serviceResult;
         }
 
+        public ServiceResult(bool successful)
+        {
+            _successful = successful;
+        }
+
+        public bool Successful()
+        {
+            return _successful ?? IsValid();
+        }
+
+        public bool IsValid()
+        {
+            return !(this.Count > 0);
```

Comment:
`Count` actually counts all elements in collection. It's much more optimal to use `Any()` method (LINQ) ;)